src/renderer/gl_engine/tvgGlCommon.h

```diff
@@ -53,6 +53,11 @@
           } \
         } while(0)
 
+static inline float getScaleFactor(const Matrix& m)
```

Comment:
just so you know, this appears to scale the size along the x-axis. The sw engine combines the two coordinate components to apply an approximate transformation in both cases, such as:
```
sqrt(transform.e11 * transform.e11 + transform.e12 * transform.e12)
```

Reply:
If the x-axis and y-axis scaling ratios are inconsistent, the drawn content will be stretched.
I just don't want to consider this special case for now. :(

Reply:
@RuiwenTang sure. let's push it.